packages/config-resolver/src/components.ts

```diff
@@ -92,9 +92,19 @@ export interface AWSClientRuntimeConfiguration extends RuntimeDependencies {
   signingName?: string;
 
   /**
-   * The service name with which to construct endpoints.
+   * The name of the service. By default same to endpoint prefix
    */
   service?: string;
+
+  /**
+   * The function that provides the default endpoint
+   */
+  endpointProvider?: (region: string, options?: any) => HttpEndpoint;
+
+  /**
+   * The function that provides the default region used to create the signer
+   */
+  signingRegionProvider?: (region: string) => string;
```

Comment:
Why is "signingRegionProvider" necessary? Why is this not part of the endpointProvider? My PR here has an example of what I'm talking about, and I think it results in a simpler system with fewer things to worry about configuring: https://github.com/aws/aws-sdk-js-v3/pull/472